compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -2586,6 +2586,10 @@ class TypeComparer(@constructorOnly initctx: Context) extends ConstraintHandling
         provablyDisjoint(tp1, gadtBounds(tp2.symbol).hi) || provablyDisjoint(tp1, tp2.superType)
       case (tp1: TermRef, tp2: TermRef) if isEnumValueOrModule(tp1) && isEnumValueOrModule(tp2) =>
         tp1.termSymbol != tp2.termSymbol
+      case (tp1: TermRef, tp2: TypeRef) if isEnumValueOrModule(tp1) && !tp1.symbol.moduleClass.derivesFrom(tp2.classSymbol) =>
```

Comment:
I don't think this works correctly because enum values would return NoSymbol for `moduleClass`

Reply:
Good point, we should use `tp1.classSymbol` here instead of `tp1.symbol.moduleClass`.

Reply:
Actually `tp1.classSymbol` does not work either, as enum values don't take the type of the anonymous class.

Reply:
That seems to contradict the design of `enum`s: if the explicitly extended parent classes are not visible, then why allow them in syntax?